docs/userguide/layout.md

```diff
@@ -1,293 +1,296 @@
 # Layout
-The most prominent objects in Elsie are boxes. They serve as a layout mechanism
-and the way how to put a content into a slide.
-
-### Creating boxes
-
-Let us create the following example, where 3 boxes are created from the
-top-level box. The new box is created by calling `.box(...)` on an existing box.
-The parameter ``slide`` passed into the slide function is also a box.
-
-```python
-@elsie.slide()
-def boxdemo1(slide):
+*Elsie* contains a layout system which allows you to quickly build scenarios that are common in
+presentations, while still providing the option of finely tuned customization for situations where
+every pixel placement matters.
+
+The central element of the layout system is the **Box**.
+
+Each *Elsie* slide contains a layout hierarchy tree. The internal nodes of the tree are *boxes* and
+the leaves are *box items*. Boxes are layout containers, which do not produce any visual content, but
+they dictate how are their children laid out on a slide. Box items are individual paintable items,
+such as text, images, shapes, etc. Anything that can be rendered by *Elsie* thus has an accompanied
+parent box which decides its size and position on a slide.
+
+## Creating boxes
+To create a new box, you can call the [`box`](elsie.boxmixin.BoxMixin.box) method on an existing
+[`Box`](elsie.boxmixin.BoxMixin). This will return a new box which will be a child of the box object
+on which you call the `box` method. The root box of the slide layout hierarchy is the
+[`Slide`](elsie.slidecls.Slide) object itself, therefore you can create other boxes by calling
+`box` on it.
+
+Here we create three boxes as children of the top-level slide box and create a child text item
+in each box.
+```elsie,width=200,height=200,skip=2
+@slides.slide()
+def three_boxes(slide):
     slide.box().text("Box 1")
     slide.box().text("Box 2")
     slide.box().text("Box 3")
 ```
 
-The code will create the following slide:
-
-<img width="512px" height="384px" src="slide_imgs/boxdemo0.png">
-
-Boxes do not produce any visible content but it influence positions of other
-elements. To see where the boxes are, we can switch on the box debug mode:
-
-```python
-@elsie.slide(debug_boxes=True)
-def boxdemo1(slide):
+The boxes themselves are invisible, but they have caused the three text items to be rendered
+below one another. If you are fine-tuning or debugging the layout of your slide and you want to see
+the extents and bounds of your boxes, you can use the
+[`debug_boxes`](elsie.slidecls.Slide.__init__) parameter when creating a slide:
```

Comment:
Maybe better link to slide decorator? (or new_slide function). Users should not call Slide constructor directly